lib/bannerFactory.js

```diff
@@ -33,16 +35,33 @@ class BannerFactory extends BaseFactory {
      * @param {String}  config.createdBy           The username of the associated user
      * @param {Boolean} [config.isActive=false]    Whether the banner is active
      * @param {String}  [config.type='info']       Type of banner (info|warn|etc)
+     * @param {String}  [config.scope='GLOBAL']    Scope of the banner (GLOBAL|PIPELINE|BUILD)
      * @memberof BannerFactory
      */
-    create(config) {
+    async create(config) {
         if (!config.type) {
             config.type = 'info';
         }
         if (!config.isActive) {
             config.isActive = false;
         }
 
+        const scopeFactories = {
+            PIPELINE: PipelineFactory,
+            BUILD: BuildFactory
+        };
+
+        if (scopeFactories[config.scope]) {
```

Comment:
Other than `GLOBAL`, we need to ensure that we validate the `scopeId`
```suggestion
        if (config.scope !== 'GLOBAL') {
```